armi/reactor/converters/axialExpansionChanger/expansionData.py

```diff
@@ -28,82 +28,103 @@
     Flags.SLUG,
 ]
 
+if TYPE_CHECKING:
+    from armi.reactor.components import Component
+    from armi.reactor.blocks import Block
+    from armi.reactor.assemblies import Assembly
 
-def getSolidComponents(b):
+
+def iterSolidComponents(b: "Block") -> Iterable["Component"]:
+    """Iterate over all solid components in the block."""
+    return filter(lambda c: not isinstance(c.material, material.Fluid), b)
+
+
+def getSolidComponents(b: "Block") -> list["Component"]:
```

Comment:
Just pulled down your branch. I see they aren't actually strings. Or at least not rendering as such in vscode. I suppose this is what you're going for? But also odd since `Block` and such have been defined already. Maybe I'm off? 
![image](https://github.com/user-attachments/assets/d1679d7f-1985-43c7-a262-545fd55e4bfe)

Reply:
> I see they aren't actually strings. Or at least not rendering as such in vscode

> But also odd since Block and such have been defined already

They're type hints for things not included in the **runtime** environment. The `Component` and `Block` definitions are imported under this [`typing.TYPE_CHECKING` guard](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING)
```python
if TYPE_CHECKING:
    from armi.reactor.components import Component
    from armi.reactor.blocks import Block
    from armi.reactor.assemblies import Assembly
```
so importing this file is not going to import `Component` nor `Block` into your environment. But type checkers, like pylance from vscode, **do** set `TYPE_CHECKING` to be truthy, so they import those objects so we developers can do type hinting on helpful things.

I try to put things that I only need to import for type checking under this guard to avoid the (sometimes small, sometimes not small) hit of importing modules at run time just for type checking. We don't _really_ need to dig into `armi.reactor.blocks` and bring out `Block` at run time for this module to work.

Happy to go either way, just let me know